src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets

```diff
@@ -421,18 +421,16 @@ Copyright (c) .NET Foundation. All rights reserved.
   <Target Name="ResolveOffByDefaultAnalyzers" AfterTargets="ResolveTargetingPackAssets"
           Condition="'@(FrameworkReference)' != ''">
     <ItemGroup>
-      <OffByDefaultAnalyzer Include="Microsoft.AspNetCore.Http.RequestDelegateGenerator"
+      <OffByDefaultAnalyzer Include="Microsoft.AspNetCore.Http.RequestDelegateGenerator.dll"
```

Comment:
I don't have a strong opinion but I would not encode the file extension here, especially as I could imagine customers adding to it. Adding the extension feels like a workaround for a slightly more complex implementation.

Reply:
@captainsafia @dsplaisted any strong opinion here? I'd really like to get this change in so we have some wiggle room before preview 4 snaps.

Reply:
This is fine with me, I don't think `OffByDefaultAnalyzer` is something we expect customers to add to.

If you did want to avoid requiring `.dll` on it though, you could probably run it through an item transform to add the `.dll` automatically.